### ISSUE TITLE
0.58.0 errors on grants

### ISSUE THREAD

**Issue by bennylu2** [Snowflake-Labs/terraform-provider-snowflake#1601]
**Provider Version**

0.58.0

**Terraform Version**

1.3.4

**Describe the bug**

Grant resources failing on `unexpected number of ID parts`

The provider is not crashing but throws the same number of errors related to #1588

```sh
│ Error: unexpected number of ID parts (1), expected 6
│ 
│   with module.tdm_demo.module.public_schemas["SUMMARIZED"].snowflake_schema_grant.this["CREATE TABLE"],
│   on .terraform/modules/tdm_demo/tdm/modules/schema/grants.tf line 84, in resource "snowflake_schema_grant" "this":
│   84: resource "snowflake_schema_grant" "this" {

│ Error: unexpected number of ID parts (1), expected 6
│ 
│   with module.tdm_demo.module.public_schemas["SUMMARIZED"].snowflake_schema_grant.this["CREATE VIEW"],
│   on .terraform/modules/tdm_demo/tdm/modules/schema/grants.tf line 84, in resource "snowflake_schema_grant" "this":
│   84: resource "snowflake_schema_grant" "this" {

...

│ Error: unexpected number of ID parts (1), expected 7
│ 
│   with module.tdm_demo.module.private_raw.snowflake_view_grant.this["SELECT"],
│   on .terraform/modules/tdm_demo/tdm/modules/schema/grants.tf line 119, in resource "snowflake_view_grant" "this":
│  119: resource "snowflake_view_grant" "this" {
│ 
╵
Operation failed: failed running terraform plan (exit 1)
```

lock file

```hcl
provider "registry.terraform.io/snowflake-labs/snowflake" {
  version     = "0.58.0"
  constraints = ">= 0.35.0, >= 0.40.0, >= 0.49.0, ~> 0.49"
  hashes = [
    "h1:grmiQruriv8kj/CRFD9tZP26ivVbV3aF3G2q7Dpk40M=",
    "zh:1de375fbe498b60adda9eb158ac1893168fdc75b627f97135d80ac0dd409e817",
    "zh:277fd8cbb2b48e08bd9381ffdc6a85fbeae91403423ed745545550e89b46df81",
    "zh:4e03ea5b81c624574f64153e37a3154e4d65fc0bd8bca673ebe65969105ca103",
    "zh:5064f4c0779a692e11fc433ea7225faafafb7fed8460b7c1073602bda1a83c24",
    "zh:5b71741cf855d33e764ac372584721a41256668e88397b4ad66b24cfc832da9c",
    "zh:64e53966e8d9dd8eefe662f82b3d557a4e34dc2254063e7eebfe6c7d3050711b",
    "zh:6bf19c018bd24b804e7087e9f4f1fd63f94a6e57d80c4d64a721929899d23dee",
    "zh:76f38ea52c6805874699133471e12e618748ed5ab44a0046763aece054c95dd4",
    "zh:873c8a2161bae696b10645c0fa376c86ca0e6f7c70bc38bca21f1caa72ff70df",
    "zh:a0f87ce426fcce6aa2d13e72dbfa5d526112cbc330f98512ffaeb063c42b1492",
    "zh:a4674e946dddd4b67d2e3284d18e55e215833f872f9553e73e18d6552729e419",
    "zh:bf24ab98f486d1675445bbb1efd3d89f92bfabc8ad93e37e0fb0d2766de94d47",
    "zh:c225996ad0874b0f130dada1df66dda829d63112ebbf73c8c1b3e0a516d1fb89",
    "zh:c4a73a8be09a52fdf843e88dba5e9ecdb5aca0c7086025e5ccecb15c7c3c8412",
    "zh:cc50667ca2864f31469d5a380ecba50f0588acb8292f821b7518e4056d97230f",
    "zh:d0802ae316bd8a2c0cd52462f15e52eca6be5d750d9245edf44111edc8bc89f4",
    "zh:e2b7f5db538798dc048701f8b703c7a604814b65cb14d1f0c2609ff99b8109a1",
    "zh:f569b65999264a9416862bca5cd2a6177d94ccb0424f3a4ef424428912b9cb3c",
  ]
}
```

**Expected behavior**

Correct number of ID parts - no errors

Likely introduced with #1598

**Comment by bennylu2** [comment]
state show

```sh
tf state show 'module.tdm_demo.module.public_schemas["SUMMARIZED"].snowflake_schema_grant.this["CREATE TABLE"]'                                                                       
# module.tdm_demo.module.public_schemas["SUMMARIZED"].snowflake_schema_grant.this["CREATE TABLE"]:
resource "snowflake_schema_grant" "this" {
    database_name          = "TDM_DEMO"
    enable_multiple_grants = true
    id                     = "TDM_DEMO|SUMMARIZED||CREATE TABLE|false"
    on_future              = false
    privilege              = "CREATE TABLE"
    roles                  = [
        "TDM_DEMO__SUMMARIZED__WRITE",
    ]
    schema_name            = "SUMMARIZED"
    shares                 = []
    with_grant_option      = false
}
```

configuration

```terraform
resource "snowflake_schema_grant" "this" {
  for_each = transpose(local.grants_schema)

  database_name          = snowflake_schema.this.database
  schema_name            = snowflake_schema.this.name
  privilege              = each.key
  roles                  = each.value
  enable_multiple_grants = true # needed until we sunset database migrations
  shares                 = []
}

```

Similarly reported in #1588

**Comment by Bryan-Meier** [comment]
I also have this same issue with snowflake_warehouse_grant:
Error: unexpected number of ID parts (5), expected 4

**Comment by joshtree41** [comment]
I think there are a few issues that track this going back to 0.56.3 or maybe 0.56.4.

**Comment by alldoami** [comment]
I'm also now getting this error 😢

**Comment by cablespaghetti** [comment]
This is affecting us in a big way too. Would be good to see it fixed 👍 

> I think there are a few issues that track this going back to 0.56.3 or maybe 0.56.4.

I searched for this error and can't find it anywhere else on the bug tracker. Can you link the related issues?

**Comment by joshtree41** [comment]
> I searched for this error and can't find it anywhere else on the bug tracker. Can you link the related issues?

I think it started [here](https://github.com/Snowflake-Labs/terraform-provider-snowflake/issues/1588) I thought there was 
another one but now i can't find it. For me the issues with grants 0.56.4 - before that I don't get the error, after that I do.

**Comment by bennylu2** [comment]
> This is affecting us in a big way too. Would be good to see it fixed 👍
> 
> > I think there are a few issues that track this going back to 0.56.3 or maybe 0.56.4.
> 
> I searched for this error and can't find it anywhere else on the bug tracker. Can you link the related issues?

It started with version `0.56.3` #1514 

Subsequent releases had provider crash issues. Now it is errors so progress!

#1547
#1588

**Comment by joshtree41** [comment]
Has there been any movement on this?

**Comment by sfc-gh-swinkler** [comment]
Sorry I was on vacation. FIx released as part of 0.58.2. Please open a new issue if you have further problems, I will be happy to help with this.

**Comment by bennylu2** [comment]
Opened #1627
